R/load_latest_forecasts_repo.R

```diff
@@ -42,49 +48,57 @@ load_latest_forecasts_repo <- function(file_path,
     models <- all_valid_models
   }
   
+  # get valid location codes
+  if (hub[1] == "US") {
+    valid_location_codes <- covidHubUtils::hub_locations$fips
+  } else if (hub[1] == "ECDC") {
+    valid_location_codes <- covidHubUtils::hub_locations_ecdc$location
+  }
+  
   # validate locations
   if (!is.null(locations)){
-    all_valid_fips <- covidHubUtils::hub_locations$fips
-    locations <- match.arg(locations, choices = all_valid_fips, several.ok = TRUE)
-  } 
+    locations <- match.arg(locations, choices = valid_location_codes, several.ok = TRUE)
+  } else {
+    locations <- valid_location_codes
+  }
   
   # validate types
   if (!is.null(types)){
     types <- match.arg(types, choices = c("point", "quantile"), several.ok = TRUE)
-  } 
+  } else {
+    types <- c("point", "quantile")
+  }
   
   # validate targets
+  # set up Zoltar connection
+  zoltar_connection <- setup_zoltar_connection()
+  # get Zoltar project url
+  project_url <- get_zoltar_project_url(hub = hub, 
+                                        zoltar_connection = zoltar_connection)
+  # get valid targets
+  all_valid_targets <- zoltr::targets(zoltar_connection, project_url)$name
   if (!is.null(targets)){
-    # set up Zoltar connection
-    zoltar_connection <- zoltr::new_connection()
-    if(Sys.getenv("Z_USERNAME") == "" | Sys.getenv("Z_PASSWORD") == "") {
-      zoltr::zoltar_authenticate(zoltar_connection, "zoltar_demo","Dq65&aP0nIlG")
-    } else {
-      zoltr::zoltar_authenticate(zoltar_connection, Sys.getenv("Z_USERNAME"),Sys.getenv("Z_PASSWORD"))
-    }
-    
-    # construct Zoltar project url
-    the_projects <- zoltr::projects(zoltar_connection)
-    project_url <- the_projects[the_projects$name == "COVID-19 Forecasts", "url"]
-    
-    # validate targets 
-    all_valid_targets <- zoltr::targets(zoltar_connection, project_url)$name
-    
     targets <- match.arg(targets, choices = all_valid_targets, several.ok = TRUE)
+  } else {
+    targets <- all_valid_targets
   }
```

Comment:
If the `targets` is `NULL`, can we just leave it as `NULL` with the behavior later on being to keep all targets if an explicit list of targets to keep is not provided?  This is the current behavior in `load_forecast_files_repo`.

A suggested revision here is along these lines, moving the interaction with Zoltar into the if block and deleting the else block:

```
  # validate targets
  if (!is.null(targets)){
    # set up Zoltar connection
    zoltar_connection <- setup_zoltar_connection()

    # get Zoltar project url
    project_url <- get_zoltar_project_url(hub = hub, 
                                        zoltar_connection = zoltar_connection)

    # get valid targets
    all_valid_targets <- zoltr::targets(zoltar_connection, project_url)$name

    # validate targets 
    all_valid_targets <- zoltr::targets(zoltar_connection, project_url)$name

    targets <- match.arg(targets, choices = all_valid_targets, several.ok = TRUE)
  }
```

Reply:
maybe we don't need the zoltar connection at all? The locations are handled by `load_forecast_files_repo()` in line [196](https://github.com/reichlab/covidHubUtils/blob/310949cecbe657d7bd3b1e1f217cd4099ca2eb63/R/load_latest_forecasts_repo.R#L196). We could change that line to give a warning if a certain requested location isn't there. But avoiding the zoltar validation would save users some time

Reply:
so maybe we could just delete that code block for now and let the targets argument get directly passed on to the next function

Reply:
I'd rather avoid issuing warnings if a location or target is not available in a given forecast since not all models submit for all locations and targets; we'd end up with a lot of warnings on loading routine forecasts.

I agree that it would be good to eliminate calls to Zoltar, but it does also seem helpful to validate these. This is something that was handled differently before and then we explicitly moved to querying Zoltar for them -- and I can't quite remember why or what the discussion was around that. It seems like it might be good to check in with others before removing this or switching back to locally stored lists of reference targets to validate against.

How about if for the short term, to be able to merge this PR sooner, we go ahead with the suggested revision above which does validations when necessary and avoids the cost of querying Zoltar when not necessary, and we can revisit whether to query Zoltar at all in a separate issue?